developer-docs-site/docs/nodes/staking.md

```diff
@@ -0,0 +1,236 @@
+
+If you do not extend your lock up period, then the remaining lockup period will decrease linearly over time, eventually becoming zero at the end of the two years. In this case, after the two years have elapsed your lockup period is zero and hence you will no longer receive any rewards.
+
+### Rewards based on the voting performance
+
+Your rewards calculation also uses your voting performance. Once you are in the validator set, you can vote in every epoch. The more consistently you vote, i.e., vote in every epoch, without any missed votes, you will receive additional voting power. This voting power is used to calculate your rewards. 
+
+For every epoch, your voting performance is determined as follows:
+
+- A running count of your missed votes, `validator_missed_votes_counts`, is maintained.
+- The number of successful votes cast by you is calculated as:
+
+```
+Total number of successful votes = Total number of blocks in the epoch - Total number of your missed votes in the current epoch
+```
+
+Hence:
+
+```
+Reward = rewards_rate * (Number of successful votes / Total number of blocks in the current epoch)
+```
+
+:::tip
+A validator’s missed votes count does not affect whether the validator is in the validator set or not. The missed votes count is used only to calculate the rewards, using the above formula.
+:::
+
+### Maintaining high rewards
+
+You can prevent your rewards from gradually declining by regularly extending your lockup period. You can extend or renew your lockup period any time in a permissionless way.
+
+For example, if you locked up for two years. A month from now you will receive a little less reward because your remaining lockup period will then be less (two years minus one month). However, if, before the month has fully elapsed, you extend your lockup period by one month to bring it back up to two years, then your month-end rewards will not decrease as they will be calculated based on the extended lockup period of two years.
+
+:::tip
+All your rewards are also subject to lockup period as they are added to the original staked amount. Hence you cannot withdraw your rewards until your lockup period has entirely expired.
+:::
+
+## Leaving the validator set
+
+You can leave the validator set in the following ways:
+
+:::tip
+See the Aptos Stake module in Move language here: [https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/configs/stake.move](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/configs/stake.move)
+:::
+
+- When your lockup period expires, you can choose not to extend it and instead call the following sequence of functions to leave the validator set:
+    - Call `Stake::unlock` to unlock your stake amount.
+    - Call `Stake::withdraw` to withdraw your staked amount at the next epoch, and finally
+    - Call `Stake::leave_validator_set` to be removed from the validator set.
```

Comment:
Those steps and the diagram can cause a bit of confusion, like if someone called "unlock" and "withdraw" but didn't call "leave_validator_set", does that mean they stay in the validator set without meeting minimal stake? We should state that if their stake amount is less than min, they will automatically leave the validator set. also maybe we can put "leave_validator_set" before the unlock and withdraw?